src/main/java/core/basesyntax/Square.java

```diff
@@ -0,0 +1,21 @@
+package core.basesyntax;
+
+public class Square extends Figure {
+    private double firstLeg;
+
+    public Square(String color, double firstLeg) {
+        super(color);
+        this.firstLeg = firstLeg;
+    }
+
+    public double calculateArea() {
+        return firstLeg * firstLeg;
```

Comment:
The `calculateArea()` method should be part of an `AreaCalculator` interface, as per the checklist. The `Square` class should implement this interface to adhere to the single responsibility principle.